text/0001-remove_ambiguous_grammar.md

```diff
@@ -0,0 +1,234 @@
+# SurrealQL: Remove ambiguity in grammar
+
+Status: Open for suggestion
+Creator: Mees Delzenne
+Owner: Mees Delzenne
+Last edited time: October 11, 2023 3:56 PM
+tag: SurrealQL
+
+# 1. Summary
+
+The current version of SurrealQL grammar, as defined by what the parser currently accepts, contains several ambiguous productions. These productions are parsed differently depending on the context or can seem very similar to each other, but subtle differences can result in completely different semantics.
+
+These ambiguities in the grammar complicate parser design, limit possible future extensions to SurrealQL that don't break, and could be confusing when using the language.
+
+This RFC proposes several changes to the grammar to limit the present ambiguity:
+
+- Disallow a value from starting with an identifier which could, in its current position in the grammar, start a statement as a keyword.
+- Require that raw identifiers don't start with a digit.
+- Introduce strand prefixes for the specific types of strands
+- Introduce a syntax error for block record-id object ambiguity
+- Change the KNN operator from `<3>` to `knn<3>` or similar
+
+## Glossary
+
+- Production: A branch or leaf of the parse tree.
+- Identifier: Names used in code which don't belong to the structure of a statement.
+- Raw Identifier: A identifier without being surrounded by ` or `⟨ ⟩`
```

Comment:
```suggestion
- Raw Identifier: A identifier without being surrounded by \` or `⟨ ⟩`
```

Reply:
Seems like I was looking at an old commit, I don't see the option to resolve however